manifests/init.pp

```diff
@@ -211,12 +211,12 @@
   String $version = $foreman::params::version,
   Enum['installed', 'present', 'latest'] $plugin_version = $foreman::params::plugin_version,
   Boolean $db_manage = $foreman::params::db_manage,
-  Optional[Stdlib::Host] $db_host = 'UNSET',
+  Stdlib::Host $db_host = 'UNSET',
   Variant[Undef, Enum['UNSET'], Stdlib::Port] $db_port = 'UNSET',
-  Optional[String] $db_database = 'UNSET',
-  Optional[String] $db_username = $foreman::params::db_username,
-  Optional[String] $db_password = $foreman::params::db_password,
-  Optional[String] $db_sslmode = 'UNSET',
+  String $db_database = 'UNSET',
+  String $db_username = $foreman::params::db_username,
+  String $db_password = $foreman::params::db_password,
+  String $db_sslmode = 'UNSET',
```

Comment:
this seems to have broken nightly:

```
[2022-06-24T07:59:45.259Z] [0;31m    2022-06-24 07:59:45 [ERROR ] [root] Parameter foreman-db-host invalid:  is not one of regexes matching /\A(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])\z/ or regexes matching /^((([0-9](?!\d)|[1-9][0-9](?!\d)|1[0-9]{2}(?!\d)|2[0-4][0-9](?!\d)|25[0-5](?!\d))[.]){3}([0-9](?!\d)|[1-9][0-9](?!\d)|1[0-9]{2}(?!\d)|2[0-4][0-9](?!\d)|25[0-5](?!\d)))(\/((([0-9](?!\d)|[1-9][0-9](?!\d)|1[0-9]{2}(?!\d)|2[0-4][0-9](?!\d)|25[0-5](?!\d))[.]){3}([0-9](?!\d)|[1-9][0-9](?!\d)|1[0-9]{2}(?!\d)|2[0-4][0-9](?!\d)|25[0-5](?!\d))|[0-9]+))?$/ or regexes matching /\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?\s*$/[0m
[2022-06-24T07:59:45.259Z] [0;31m    2022-06-24 07:59:45 [ERROR ] [root] Parameter foreman-db-database invalid: nil is not a valid string[0m
[2022-06-24T07:59:45.259Z] [0;31m    2022-06-24 07:59:45 [ERROR ] [root] Parameter foreman-db-sslmode invalid: nil is not a valid string[0m
```
```

Reply:
I forgot Kafo treats `'UNSET'` as undef: https://github.com/theforeman/kafo/blob/3fc94d16f9d92d810bf2bde9eff05eb5a19a15f7/lib/kafo/param.rb#L7=

Perhaps it's time to move it to a just `undef` instead of this hack.

Reply:
https://github.com/theforeman/puppet-foreman/pull/1066 does that.